subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -75,8 +75,11 @@ static int download_client_callback(const struct download_client_evt *event)
 		if (err != 0) {
 			LOG_ERR("dfu_target_write error %d", err);
 			err = download_client_disconnect(&dlc);
+			if (err != 0) {
+				LOG_ERR("Unable to disconnect download_client");
+			}
 			callback(FOTA_DOWNLOAD_EVT_ERROR);
-			return err;
+			return -EFAULT;
```

Comment:
Before if the write failed we would disconnect which would set `err = 0` this would make `return 0` which would allow the `download_client` to continue in it's `while` loop instead of breaking triggering a 2nd event  `DOWNLOAD_CLIENT_EVT_DONE`

Reply:
The log seems a bit redundant, `download_client_disconnect` logs errors already but up to you :)
Also I would simply return `-1`, there is no need to return any specific `errno`.

Reply:
I just left it as is to reduce the amount of changes..